manifests/controller/config.pp

```diff
@@ -89,134 +64,142 @@
-  Optional[Enum['human','json']]                       $format             = undef,
-  Optional[String]                                     $hiera_config       = undef,
-  Optional[String]                                     $inventoryfile      = undef,
-  Optional[Enum['docker','local','pcp','ssh','winrm']] $transport          = undef,
-  Optional[Boolean]                                    $ssh_host_key_check = undef,
-  Optional[String]                                     $ssh_password       = undef,
-  Optional[Integer[0]]                                 $ssh_port           = undef,
-  Optional[String]                                     $ssh_private_key    = undef,
-  Optional[String]                                     $ssh_proxyjump      = undef,
-  String                                               $ssh_tmpdir         = $simp_bolt::target::user_home,
-  String                                               $ssh_user           = $simp_bolt::target::user_name,
-  String                                               $ssh_run_as         = $simp_bolt::target::user_sudo_user,
-  Enum['debug', 'info', 'notice', 'warn', 'error']     $log_console_level  = 'info',
-  String                                               $log_file           = '/var/log/puppetlabs/bolt/bolt.log',
-  Enum['debug', 'info', 'notice', 'warn', 'error']     $log_file_level     = 'info',
-  Optional[Boolean]                                    $log_file_append    = undef,
-  Boolean                                              $disable_analytics  = true,
+  # Local Target Directory Options
+  Optional[String[1]]              $local_group        = $simp_bolt::controller::local_group_name,
+  Stdlib::Unixpath                 $local_home         = pick($simp_bolt::controller::local_user_home, '/var/local/simp_bolt'),
+  Optional[String[1]]              $local_user         = $simp_bolt::controller::local_user_name,
+
+  # Config File Specification
+  Optional[Hash]                   $config_hash        = undef,
+
+  # Global Bolt Options
+  Boolean                          $color              = true,
+  Optional[Integer[0]]             $concurrency        = undef,
+  Simp_bolt::Transport             $default_transport  = 'ssh',
+  Boolean                          $disable_analytics  = true,
+  Optional[Enum['human','json']]   $format             = undef,
+  Optional[String[1]]              $hiera_config       = undef,
+  Optional[String[1]]              $inventoryfile      = undef,
+  Simp_bolt::LogLevel              $log_console_level  = 'info',
+  Boolean                          $log_file_append    = false,
+  Simp_bolt::LogLevel              $log_file_level     = 'info',
+  Stdlib::Unixpath                 $log_file           = '/var/log/puppetlabs/bolt/bolt.log',
+  Optional[String[1]]              $modulepath         = undef,
+
+  # Overall Transport Options
+  Hash[Simp_bolt::Transport, Hash] $transport_options  = {
+                                     'ssh' => {
```

Comment:
Why not match the parameters specified in `simp_bolt::target` for `user_name` and `user_home`?  I can see not pointing to the target manifest so the classes can be decoupled but then I think the parameters should be specified in the init manifest and referenced from there.  That way they match by default but can be overwritten in hiera for either class if desired.